### ISSUE TITLE
Mono gets stuck when attempting to move to current energy

### ISSUE THREAD

**Issue by prjemian** [epics-modules/optics#9]
From APS 4-ID-C Polar (https://github.com/APS-4ID-POLAR/ipython-polar/issues/52#issuecomment-747586392),

> [the EPICS monochromator code] gets stuck in moving if you try to move to the same energy.

See this animated graphic to reproduce:

![test](https://user-images.githubusercontent.com/10960331/102524444-9ffa0c00-405e-11eb-95f8-cb44d25f12c7.gif)

Pressing the `Move` button unsticks it.  Is there a way to fix in the monochromator support code here?  The current behavior jams up Bluesky control.  (Bluesky *could* add a check if the move is trivial and then avoid asking EPICS to set the energy to the same value.)

**Comment by gfabbris** [comment]
Should we ask somebody in particular about this?

**Comment by mdwyman** [comment]
Both the MOVE button and the BraggEAO PVs have a forward link that sets the Mono's busy record to "Moving".  However, the SNL program is where the busy records is "un-set" (i.e. to "Done").  The Move button activation does trigger it's associated SNL program variable so eventually the busy record completes.  Since the Energy isn't changed, it doesn't trigger the SNL monitor, so the the busy record is never completed.  

If the MDEL field of the BraggEAO PV is set to -1, the SNL monitor will trip.  MDEL is the monitor deadband and by default it's set to 0, so monitors are only processed if value is changed.  When set to -1, monitors are processed every time the record is processed not just when the value is changed.

**Comment by mdwyman** [comment]
Not sure what the other repercussions of setting MDEL to -1 are, but since it seems to only affect Bluesky/ophyd operations, maybe part of the bluesky session startup/instrument initialization be to set MDEL = -1?

**Comment by prjemian** [comment]
I can check in my simulator/softIoc.

**Comment by prjemian** [comment]
I was able to reproduce the mono getting stuck with a softIoc without using Bluesky.

- Set to **Auto**
- set energy to 5.0
- wait
- for move to complete (`IOC:KohzuMoving` is clear)
- set energy to 5.0 (again)
- `IOC:KohzuMoving` is stuck
- Press **Move** button
- `IOC:KohzuMoving` is clear

**Comment by prjemian** [comment]
Should resolve this in the sequence program.

**Comment by mdwyman** [comment]
> I was able to reproduce the mono getting stuck with a softIoc without using Bluesky.
> 
> * Set to **Auto**
> * set energy to 5.0
> * wait
> * for move to complete (`IOC:KohzuMoving` is clear)
> * set energy to 5.0 (again)
> * `IOC:KohzuMoving` is stuck
> * Press **Move** button
> * `IOC:KohzuMoving` is clear

Yes, but how is setting IOC:BraggEAO.MDEL = -1 not a viable solution? 

I was also able to get similar behavior for the Lambda and Theta positioners: set lambda (or theta) value, then set it to the same value, system gets stuck in Moving mode; **Move** clears it. Again setting MDEL = -1 for the positioners seems to fix.

With MDEL = 0, the monitor is never set when the energy (or lambda/theta) is set (again) to it's current value, so it wouldn't matter the method of interaction (bluesky, caqtdm, camonitor, etc).  With MDEL = -1, the monitor is updated (can be seen with camonitor)

Looking at the sequence program, it's dependent on the energy (or lambda/theta) PV setting a monitor. Changes to both the sequence program and database would be needed to flag the PV being set with an old value when in "Auto" mode.

**Comment by mdwyman** [comment]
@prjemian  Tried my hand at SNL.  Added handling of the busy signal since that's the only "change" when bluesky or user sets energy to previous value. Update here https://github.com/epics-modules/optics/commit/4ab1d65db19e07207c10cfdc6edd20424b17e832 - only change is to the kohzuCtl.st

**Comment by prjemian** [comment]
I'll take a look tomorrow

**Comment by prjemian** [comment]
> I was also able to get similar behavior for the Lambda and Theta positioners: set lambda (or theta) value, then set it to the same value, system gets stuck in Moving mode; Move clears it. Again setting MDEL = -1 for the positioners seems to fix.

Without looking more deeply, this looks like a situation where DMOV is not sent to 0 then back to 1.  Even for new moves where VAL == RBV, this is the sequence:

1. VAL field written (different value or same value)
2. DMOV goes to zero
3. MOVN goes to 1
4. motor is moved until RBV reaches target (within RDBD, as I recall) or error
5. MOVN goes to zero
6. DMOV goes to 1
